modules/gapi/src/api/gproto.cpp

```diff
@@ -197,6 +197,45 @@ bool cv::can_describe(const GMetaArgs &metas, const GRunArgs &args)
                      });
 }
 
+bool cv::validate_input_arg(const GRunArg& arg)
+{
+    // FIXME: It checks only Mat's argument
+    switch (arg.index())
+    {
+#if !defined(GAPI_STANDALONE)
+    case GRunArg::index_of<cv::Mat>():
+    {
+        const auto desc = descr_of(util::get<cv::Mat>(arg));
+        GAPI_Assert(desc.size.height != 0 && desc.size.width != 0 && "incorrect dimensions of cv::Mat!"); return true;
+    }
+    case GRunArg::index_of<cv::UMat>():
+    {
+        const auto desc = descr_of(util::get<cv::UMat>(arg));
+        GAPI_Assert(desc.size.height != 0 && desc.size.width != 0 && "incorrect dimensions of cv::UMat!"); return true;
+    }
+    case GRunArg::index_of<cv::Scalar>(): return true;
+#endif //  !defined(GAPI_STANDALONE)
+    case GRunArg::index_of<cv::gapi::own::Mat>():
+    {
+        const auto desc = descr_of(util::get<cv::gapi::own::Mat>(arg));
+        GAPI_Assert(desc.size.height != 0 && desc.size.width != 0 && "incorrect dimensions of own::Mat!"); return true;
+    }
+    case GRunArg::index_of<cv::gapi::own::Scalar>(): return true;
+    case GRunArg::index_of<cv::detail::VectorRef>(): return true;
+    case GRunArg::index_of<cv::detail::OpaqueRef>(): return true;
+    case GRunArg::index_of<cv::gapi::wip::IStreamSource::Ptr>(): return true;
+    default: util::throw_error(std::logic_error("Unsupported GRunArg type"));
```

Comment:
Actually, if a type doesn't require any validation, you can just do
```c++
default:
    // No extra handling
    break;
```

Reply:
Fixed